src/components/Auth/auth-method/AuthMethodSettings.tsx

```diff
@@ -0,0 +1,151 @@
+import React, { ReactElement, useMemo, useState } from "react";
+import { css } from "@emotion/css";
+import { useTheme2, Select } from "@grafana/ui";
+import { SelectableValue } from "@grafana/data";
+import { BasicAuth, Props as BasicAuthProps } from "./BasicAuth";
+import { ConfigSection } from "../../ConfigEditor";
+import { AuthMethod, CustomMethod, CustomMethodId } from "../types";
+
+const defaultOptions: Record<AuthMethod, SelectableValue<AuthMethod>> = {
+  [AuthMethod.BasicAuth]: {
+    label: "Basic authentication",
+    value: AuthMethod.BasicAuth,
+    description: "Authenticate with your data source username and password.",
+  },
+  [AuthMethod.CrossSiteCredentials]: {
+    label: "Enable cross-site access control requests",
+    value: AuthMethod.CrossSiteCredentials,
+    description:
+      "Allow cross-site Access-Control requests with your existing credentials and cookies. This enables the server to authenticate the user and perform authorized requests on their behalf on other domains.",
+  },
+  [AuthMethod.OAuthForward]: {
+    label: "Forward OAuth Identity",
+    value: AuthMethod.OAuthForward,
+    description:
+      "Forward the OAuth access token (and if available: the OIDC ID token) of the user querying to the data source.",
+  },
+  [AuthMethod.NoAuth]: {
+    label: "No Authentication",
+    value: AuthMethod.NoAuth,
+    description: "Data source is available without authentication",
+  },
+};
+
+type Props = {
+  selectedMethod: AuthMethod | CustomMethodId;
+  mostCommonMethod?: AuthMethod | CustomMethodId;
+  visibleMethods: (AuthMethod | CustomMethodId)[];
+  customMethods?: CustomMethod[];
+  onAuthMethodSelect: (authType: AuthMethod | CustomMethodId) => void;
+  basicAuth?: BasicAuthProps;
+};
+
+export const AuthMethodSettings: React.FC<Props> = ({
+  selectedMethod,
+  mostCommonMethod,
+  visibleMethods,
+  customMethods,
+  onAuthMethodSelect,
+  basicAuth,
+}) => {
+  const [authMethodChanged, setAuthMethodChanged] = useState(false);
+  const { colors, spacing } = useTheme2();
+  const isSingleMethodMode = visibleMethods.length === 1;
```

Comment:
I know we usually have no comments, but maybe it's worth adding some explanation what's happening here ?

Reply:
Added in 749499e77d874eb784c314b170c034ddf10d18bb. Is that what you meant?